src/components/steps/steps.tsx

```diff
@@ -0,0 +1,209 @@
+  StepsContainerStyle,
+  StepStyle,
+  StepDrawStyle,
+} from './styles';
+
+export interface StepProps {
+  steps: any[];
+  handleClick?: () => void;
+}
+
+export type StatusType = 'default' | 'selected' | 'checked' | 'error';
+
+export interface StepType {
+  label: string;
+  description?: string;
+  index?: number;
+  status?: StatusType;
+}
+
+export type LineType = 'before' | 'after';
+
+export type StepConfig = {
+  current?: number;
+  disabled?: boolean;
+  steps: StepType[];
+  clickable?: boolean;
+};
+
+export type StepCircle = {
+  status?: StatusType;
+  index?: number;
+  steps: StepType[];
+};
+
+export type StepLine = {
+  status?: StatusType;
+  index?: number;
+  steps: StepType[];
+  type: LineType;
+};
+
+export type StepDetails = {
+  label: string;
+  description?: string;
+  index?: number;
+};
+
+/* eslint-disable no-unused-vars */
+enum Status {
+  default = 'default',
+  selected = 'selected',
+  checked = 'checked',
+  error = 'error',
+}
+
+function LineDraw({ type, status, index, steps }: StepLine) {
+  const beforeStepIsChecked = (index: number = 0): boolean => {
+    return steps[index - 2] && steps[index - 2].status === Status.checked;
+  };
+
+  const afterStepIsChecked = (index: number = 0): boolean => {
+    if (status && status === Status.checked) return true;
+    return steps[index + 2] && steps[index + 2].status !== Status.default;
+  };
+
+  if (type === 'before') {
+    return (
+      <>{index !== 1 && <LineStyle bolded={beforeStepIsChecked(index)} />}</>
+    );
+  }
+
+  return (
+    <>
+      {index !== steps.length && (
+        <LineStyle bolded={afterStepIsChecked(index)} />
+      )}
+    </>
+  );
+}
+
+function StepDraw({ status, index, steps }: StepCircle) {
+  return (
+    <>
+      <StepDrawStyle>
+        <LineDraw type={'before'} status={status} index={index} steps={steps} />
+        <CircleStyle>
+          <span>
+            {status === 'checked' ? <IonIcon type={'check'} /> : index}
+          </span>
+        </CircleStyle>
+        <LineDraw type={'after'} status={status} index={index} steps={steps} />
+      </StepDrawStyle>
+    </>
+  );
+}
+
+function Details({ label, description, index }: StepDetails) {
+  return (
+    <>
+      <DetailsStyle>
+        <div className="label">{label}</div>
+        {description && (
+          <div className="description" data-testid={'description-' + index}>
+            {description}
+          </div>
+        )}
+      </DetailsStyle>
+    </>
+  );
+}
+
+function stepStatus(step: StepType, currentIndex: number): StatusType {
+  if (step.index && step.index < currentIndex) return Status.checked;
+  if (step.index && step.index === currentIndex) return Status.selected;
+  return Status.default;
+}
+
+function checkStartedStatus(step: StepType, currentIndex: number): StatusType {
+  return step.status ? step.status : stepStatus(step, currentIndex);
+}
+
+function checkNecessityToChange(currentIndex: number, limit: number): boolean {
```

Comment:
what this do?

Reply:
This checks is necessary to change steps visually, cuz if index number before or after the existentes indexes on array, it wont be considerate, once time it will be nonexistent index on steps.

Reply:
Also, I did this to pass on codeclimate test.